src/functions/initializeActivities.ts

```diff
@@ -1,86 +1,116 @@
-                        state: OrchestratorComponentState.NotStarted,
-                        message: null
-                    }
-                },
-                async: {
-                    mandatory: {
-                    },
-                    optional: {
-
-                    },
-                    status: {
-                        state: OrchestratorComponentState.NotStarted,
-                        message: null
-                    }
-                },
-                post: {
-                    mandatory: {
-                    },
-                    optional: {
-
-                    },
-                    status: {
-                        state: OrchestratorComponentState.NotStarted,
-                        message: null
-                    }
-                },
-                status: {
-                    state: OrchestratorComponentState.NotStarted,
-                    message: null
-                }
-            };
+    if (event.stages) {
+        for (const i in event.stages) {
+            getActivityForStage(i, event);
         }
         delete event.stages;
     }
     event.workflow = event.metadata.workflow;
     event.currentDate = new Date().getTime();
-    await dynamodb.put({
-        TableName: process.env.statusTable,
-        Item: event
-    }).promise();
+    await save(event);
 
     return event;
 });
+
+export function getActivityForStage(stage: string, event: OrchestratorWorkflowStatus) {
+    if (event.activities[stage]) {
+        return;
+    }
+    event.activities[stage] = {
+        pre: {
+            mandatory: {
+            },
+            optional: {
+
+            },
+            status: {
+                state: OrchestratorComponentState.NotStarted,
+                message: null
+            }
+        },
+        async: {
+            mandatory: {
+            },
+            optional: {
+
+            },
+            status: {
+                state: OrchestratorComponentState.NotStarted,
+                message: null
+            }
+        },
+        post: {
+            mandatory: {
+            },
+            optional: {
+
+            },
+            status: {
+                state: OrchestratorComponentState.NotStarted,
+                message: null
+            }
+        },
+        status: {
+            state: OrchestratorComponentState.NotStarted,
+            message: null
+        }
+    };
+}
+export async function getActivity(event: OrchestratorWorkflowStatus): Promise<OrchestratorWorkflowStatus> {
+    const ret = await dynamodb.get({
+        TableName: process.env.statusTable,
+        Key: {
+            uid: event.uid,
+            workflow: event.workflow
+        }
+    }).promise();
+    if(!ret || !ret.Item) {
+        return;
+    }
+    const output = ret.Item as OrchestratorWorkflowStatus;
+    return output;
+}
+
+export async function save(event: OrchestratorWorkflowStatus): Promise<PromiseResult<PutItemOutput, AWSError>> {
```

Comment:
This should not be exported, as there's not really a good reason to expose this outside this file.